lib/Command/SetSecretVerificationStatusCommand.php

```diff
@@ -0,0 +1,150 @@
+
+	/** @var IUserManager */
+	private $userManager;
+
+	public function __construct(
+		TotpSecretMapper $secretMapper,
+		IUserManager $userManager
+	) {
+		parent::__construct();
+		$this->secretMapper = $secretMapper;
+		$this->userManager = $userManager;
+	}
+
+	protected function configure() {
+		$this
+			->setName('twofactor_totp:set-secret-verification-status')
+			->setDescription(
+				'Set secret verification of specified users or all users'
+			)
+			->addArgument(
+				'set-verified',
+				InputArgument::REQUIRED,
+				'Secret verification status to set. (true or false)'
+			)
+			->addOption(
+				'all',
+				null,
+				InputOption::VALUE_NONE,
+				'Will affects all users secrets'
+			)
+			->addOption(
+				'uid',
+				'u',
+				InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+				'The user\'s uid is used. This option can be used as --uid "Alice" --uid "Bob"'
+			);
+	}
+
+	/**
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @return int
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		try {
+			$options = $this->validateInput($input);
+		} catch (InvalidArgumentException $e) {
+			$message = $e->getMessage();
+			$output->writeln("<error>$message</error>");
+			return 1;
+		}
+		$status = $options['status'];
+		if($status === true) {
+			$commandActionString = 'verified';
+		} else {
+			$commandActionString = 'unverified';
+		}
+
+		if ($options['uids'] === 'all') {
+			$this->secretMapper->setAllSecretsVerificationStatus($status);
+			$output->writeln("<info>All the secrets are $commandActionString</info>");
+			return 0;
+		}
+		foreach ($options['uids'] as $uid) {
+			try {
+				$user = $this->userManager->get($uid);
+				if(!$user) {
+					$output->writeln("<error>User $uid does not exist</error>");
+					continue;
+				}
+				$dbSecret = $this->secretMapper->getSecret($user);
+				$dbSecret->setVerified($options['status']);
+				$this->secretMapper->update($dbSecret);
+				$output->writeln("<info>The secret of $uid is $commandActionString</info>");
+			} catch (DoesNotExistException $ex) {
+				$output->writeln("<error>User has no secret: $uid</error>");
+			}
+		}
+		return 0;
+	}
+
+	/**
+	 * @param InputInterface $input
+	 * @return array
+	 */
+	protected function validateInput(InputInterface $input) {
+		$status = $input->getArgument('set-verified');
+		if ($status === 'true') {
+			$status = true;
+		} elseif ($status === 'false') {
+			$status = false;
+		} else {
+			throw new InvalidArgumentException(
+				"Argument value is not valid. Use true or false as argument value."
+			);
+		}
+
+		if ($input->getOption('all') !== false) {
+			$uids = 'all';
+		} else if(\count($input->getOption('uid'))) {
+			$uids = $input->getOption('uid');
+		} else {
+			throw new InvalidArgumentException(
+				"No user input specified. Use at least one --uid option to specify an user or --all option for all users."
+			);
+		}
+
+		$options = [
```

Comment:
ah, here they get converted. I was confused by reading the code above and so will other devs.

This method is called `validateInput` so I'd expect it to only validate, not parse/convert. Maybe rename to `parseInput` ?